MassSpectrometry/MzSpectra/IsotopicEnvelope.cs

```diff
@@ -1,30 +1,39 @@
 ﻿using MathNet.Numerics.Statistics;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace MassSpectrometry
 {
     public class IsotopicEnvelope
     {
         public readonly List<(double mz, double intensity)> Peaks;
         public double MonoisotopicMass { get; private set; }
+        public double MostAbundantObservedIsotopicMass { get; private set; }
         public readonly int Charge;
         public readonly double TotalIntensity;
         public readonly double StDev;
         public readonly int MassIndex;
+
         public double Score { get; private set; }
 
         public IsotopicEnvelope(List<(double mz, double intensity)> bestListOfPeaks, double bestMonoisotopicMass, int bestChargeState, double bestTotalIntensity, double bestStDev, int bestMassIndex)
         {
             Peaks = bestListOfPeaks;
             MonoisotopicMass = bestMonoisotopicMass;
+            MostAbundantObservedIsotopicMass = GetMostAbundantObservedIsotopicMass(bestListOfPeaks, bestChargeState);
             Charge = bestChargeState;
             TotalIntensity = bestTotalIntensity;
             StDev = bestStDev;
             MassIndex = bestMassIndex;
             Score = ScoreIsotopeEnvelope();
         }
 
+        public double GetMostAbundantObservedIsotopicMass(List<(double mz, double intensity)>  peaks, int charge)
+        {
+            return peaks.OrderByDescending(p => p.intensity).ToList()[0].Item1 * charge;
```

Comment:
I'm not sure about this formula. I think formula for most abundant isotopic mass (IM) of a given charge state (z) should be (IM - z * hydrogen mass) / z.